src/test/java/io/codeka/gaia/e2e/DashboardPage.kt

```diff
@@ -1,28 +1,38 @@
 package io.codeka.gaia.e2e
 
+import org.openqa.selenium.By
 import org.openqa.selenium.WebDriver
 import org.openqa.selenium.WebElement
 import org.openqa.selenium.support.FindBy
+import org.openqa.selenium.support.ui.ExpectedCondition
+import org.openqa.selenium.support.ui.ExpectedConditions
+import org.openqa.selenium.support.ui.WebDriverWait
 
 class DashboardPage (val webDriver: WebDriver){
 
     init {
-        if(! webDriver.title.equals("Gaia - A terraform UI")){
+        if(! webDriver.currentUrl.endsWith("/dashboard")){
             throw IllegalStateException("This is not the dashboard page. Current page is ${webDriver.title}")
         }
     }
 
-    @FindBy(className= "modules_count")
+    fun waitForPageLoaded() {
+        val wait = WebDriverWait(webDriver, 10)
+        wait.until(ExpectedConditions.titleIs("Gaia - Dashboard"))
+        wait.until(ExpectedConditions.numberOfElementsToBe(By.className("b-overlay"), 0))
+    }
+
+    @FindBy(xpath = "(//*[contains(@class,'widget-value')])[1]")
     private lateinit var modulesCountElement: WebElement
 
-    @FindBy(className="stacks_count")
+    @FindBy(xpath = "(//*[contains(@class,'widget-value')])[2]")
```

Comment:
We should use id selectors instead of xpath ones which seem to be more difficult to maintain

Reply:
Oh yeah, I remember why I did that.
To use an id selector in this case, we have to pass the id as a prop to the component, as the value we want is located in a sub node. I thought it was too complex for a component to expose a property in its interface, just for the sake of selenium tests. So I choose to use the 'widget-value' class with a xpath selector.
There's also a bug with Kotlin in Selenium, which doesn't understand Kotlin's covariance.
In Java, Selenium can inject a `List<WebElement>`.
In Kotlin, when we declare a `List<WebElement>`, it is in fact a `List<? extends WebElement>`, and Selenium doesn't like it. I may open an issue for that.
Anyway, I'll use something else here :)

Reply:
For future reference, the problen with the `@FindBy` annotation and type covariance is here :

org.openqa.selenium.support.pagefactory.DefaultFieldDecorator#isDecoratableList

Reply:
✅